controllers/synapse/synapse_controller.go

```diff
@@ -94,35 +96,61 @@ func (r *SynapseReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	var outputConfigMap corev1.ConfigMap
+	objectMetaForSynapse := setObjectMeta(synapse.Name, synapse.Namespace, map[string]string{})
 
 	if synapse.Spec.Homeserver.ConfigMap != nil {
 		var inputConfigMap corev1.ConfigMap
 		// Get and validate homeserver ConfigMap
 		ConfigMapName := synapse.Spec.Homeserver.ConfigMap.Name
-		if err := r.Get(ctx, types.NamespacedName{Name: ConfigMapName, Namespace: synapse.Namespace}, &inputConfigMap); err != nil {
-			reason := "ConfigMap " + ConfigMapName + " does not exist in namespace " + synapse.Namespace
+		ConfigMapNamespace := r.getConfigMapNamespace(synapse, synapse.Spec.Homeserver.ConfigMap.Namespace)
+		if err := r.Get(
+			ctx,
+			types.NamespacedName{Name: ConfigMapName, Namespace: ConfigMapNamespace},
+			&inputConfigMap,
+		); err != nil {
+			reason := "ConfigMap " + ConfigMapName + " does not exist in namespace " + ConfigMapNamespace
 			if err := r.setFailedState(ctx, &synapse, reason); err != nil {
 				log.Error(err, "Error updating Synapse State")
 			}
 
-			log.Error(err, "Failed to get ConfigMap", "ConfigMap.Namespace", synapse.Namespace, "ConfigMap.Name", ConfigMapName)
+			log.Error(
+				err,
+				"Failed to get ConfigMap",
+				"ConfigMap.Namespace",
+				ConfigMapNamespace,
+				"ConfigMap.Name",
+				ConfigMapName,
+			)
 			return ctrl.Result{RequeueAfter: time.Duration(30)}, err
 		}
 
 		if err := r.ParseHomeserverConfigMap(ctx, &synapse, inputConfigMap); err != nil {
 			return ctrl.Result{RequeueAfter: time.Duration(30)}, err
 		}
 
-		synapse.Status.HomeserverConfigMapName = ConfigMapName
-		outputConfigMap = inputConfigMap
+		// Create a ConfigMap for Synapse using the inputConfigMap data as input
+		if err := r.reconcileResource(
+			ctx,
+			r.configMapForSynapseCopy,
+			&synapse,
+			&createdConfigMap,
```

Comment:
Since the code is a bit long for this file, wouldn't be a good idea bringing the createdConfigMap variable declaration closer to where it's being used?

Reply:
I followed your suggestion and moved the variable declaration closer. Note that this is used in both branch of the if/else block as follow:


```go
	var createdConfigMap corev1.ConfigMap

	if synapse.Spec.Homeserver.ConfigMap != nil {
		// createdConfigMap becomes a copy of the existing ConfigMap
	} else {
		// A new ConfigMap is created and stored in createdConfigMap
	}
```

Reply:
It's clear now.